api/db/base.js

```diff
@@ -0,0 +1,253 @@
+const logger = require('../logger')('db base model');
+const pick = require('lodash.pick');
+
+class ValidationError extends Error {
+  constructor(message) {
+    super(message);
+
+    this.error = { error: message };
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  get statusCode() {
+    return 400;
+  }
+}
+
+const defaultSyncChildren = async (
+  // The raw data that was sent into the parent, from which child
+  // objects will be synchronized
+  data,
+  // Objects mapping data fields to child data models.  The keys
+  // this object are expected field names on the raw data, while
+  // the values are the corresponding model names.
+  childModels,
+  // An object describing all known models.  Keys are model names,
+  // values are model objects.
+  allModels,
+  // An object describing how to relate children back to the parent.
+  // Should be something that can be passed to where, and used to
+  // populate new objects.  E.g., { apd_id: 7 }
+  foreignKey,
+  // Transaction object, if this is being executed in a transaction
+  // (and in this base model, it always is, so 🤷‍♂️)
+  transacting,
+  // The name of the parent model.  Only used for logging.
+  modelName
+) =>
+  Promise.all(
+    Object.keys(childModels).map(async fieldName => {
+      if (data[fieldName]) {
+        logger.silly(
+          `${modelName} | synchornize ${fieldName} with ${
+            childModels[fieldName]
+          } model`
+        );
+        const Model = allModels[childModels[fieldName]];
+        return Model.synchronize(data[fieldName], foreignKey, transacting);
+      }
+      return true;
```

Comment:
for readability (and to allow the main chunk of code to be less indented), how do u feel about reversing this conditional return? i.e.,

```
if (!data[fieldName]) return true

const Model = allModels[childModels[fieldName]];
return Model.synchronize(data[fieldName], foreignKey, transacting);
...
```